client/src/graphql/graphql.ts

```diff
@@ -0,0 +1,130 @@
+// TS version of schema.graphql (also keep in sync with graphql.py)
+import { gql } from "@apollo/client";
+
+import { Mate } from "./types";
+
+/**
+ * GQL Fragment for Sync fields
+ */
+const SYNC_FIELDS = gql`
+  fragment syncFields on Sync {
+    id
+    timestamp
+    title
+    details
+  }
+`;
+
+/**
+ * GQL Fragment for Mate fields
+ */
+const MATE_FIELDS = gql`
+  fragment mateFields on Mate {
+    id
+    name
+    syncs @include(if: $withSync) {
+      ...syncFields
+    }
+  }
+`;
+
+/**
+ * GQL Fragment for Family fields
+ */
+const FAMILY_FIELDS = gql`
+  fragment familyFields on Family {
+    id
+    sync_interval_days
+    name
+    mates @include(if: $withMate) {
+      ...mateFields
+    }
+  }
+`;
+
+/**
+ * GQL Fragment for User fields
+ */
+const USER_FIELDS = gql`
+  fragment userFields on User {
+    id
+    first_name
+    picture_url
+    email
+    unassigned_family @include(if: $withUnassignedFamily) {
+      ...familyFields
+    }
+    families @include(if: $withFamily) {
+      ...familyFields
+    }
+  }
+`;
+
+/**
+ * GQL Query for fetching all data
+ */
+export const GET_USER_DATA = gql`
```

Comment:
This allows you to get a specified depth of info from user. Not used right now, but can be a good example for fragments and directives.